File: AspNetMvcUnleashed/0672329980_source/Chapter18/VB/UnleashedBlog/UnleashedBlog/Controllers/ArchiveController.vb
Public Class ArchiveController
    Inherits Controller

    Private _repository As BlogRepositoryBase

    Public Sub New()
        Me.New(New EntityFrameworkBlogRepository())
    End Sub

    Public Sub New(ByVal repository As BlogRepositoryBase)
        _repository = repository
    End Sub

    Public Function Index(ByVal year As Integer?, ByVal month As Integer?, ByVal day As Integer?, ByVal name As String) As ActionResult
        Return View(_repository.ListBlogEntries(year, month, day, name))
    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter19/CS/UnleashedBlog/UnleashedBlog/Controllers/ArchiveController.cs
using System.Web.Mvc;
using UnleashedBlog.Models;
using UnleashedBlog.Models.EntityFramework;

namespace UnleashedBlog.Controllers
{
    public class ArchiveController : Controller
    {
        private BlogServiceBase _blogService;

        public ArchiveController()
        {
            _blogService = new BlogService(this.ModelState);
        }

        public ArchiveController(BlogRepositoryBase blogRepository)
        {
            _blogService = new BlogService(this.ModelState, blogRepository);
        }

        public ActionResult Index(int? year, int? month, int? day, string name)
        {
            return View(_blogService.ListBlogEntries(year, month, day, name));
        }

    }
}
